### PR TITLE
Create Diodes U-DFN2510-10 footprint

### PR DESCRIPTION
**Pull request by eclecticc**
Custom footprint for Diodes' U-DFN2510-10 used in D3V3XA4B10LP, which has a slightly unusual pad layout for a DFN.  Aside from the pads, the footprint is copied from the existing USON-10_2.5x1.0mm_P0.5mm footprint.
https://www.diodes.com/assets/Datasheets/D3V3XA4B10LP.pdf

image

---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the footprint(s) you are contributing
- [x] An example screenshot image is very helpful 
- [x] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [x] Check the output of the Travis automated check scripts - fix any errors as required
- [x] Give a reason behind any intentional library convention rule violation.

---

Be patient, we maintainers are volunteers with limited time and need to check your contribution against the datasheet. You can speed up the process by providing all the necessary information (see above). And you can speed up the process even more by providing a dimensioned drawing of your contribution. A tutorial on how to do that is found here: https://forum.kicad.info/t/how-to-check-footprint-correctness/9279 (This is optional!)

### CODE REVIEW

**Comment by eclecticc** [comment]
Thanks, corrected.

File: Package_DFN_QFN.pretty/Diodes_UDFN-10_1.0x2.5mm_P0.5mm.kicad_mod [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  (pad 6 smd rect (at 0.3875 1 270) (size 0.25 0.625) (layers F.Cu F.Paste F.Mask))
+  (pad 8 smd rect (at 0.4125 0 270) (size 0.45 0.575) (layers F.Cu F.Paste F.Mask))
+  (pad 5 smd rect (at -0.3875 1 270) (size 0.25 0.625) (layers F.Cu F.Paste F.Mask))
+  (pad 9 smd rect (at 0.3875 -0.5 270) (size 0.25 0.625) (layers F.Cu F.Paste F.Mask))
+  (pad 10 smd rect (at 0.3875 -1 270) (size 0.25 0.625) (layers F.Cu F.Paste F.Mask))
+  (pad 4 smd rect (at -0.3875 0.5 270) (size 0.25 0.625) (layers F.Cu F.Paste F.Mask))
+  (pad 1 smd rect (at -0.3875 -1 270) (size 0.25 0.625) (layers F.Cu F.Paste F.Mask))
+  (pad 3 smd rect (at -0.35 0 270) (size 0.45 0.7) (layers F.Cu F.Paste F.Mask))
+  (pad 2 smd rect (at -0.3875 -0.5 270) (size 0.25 0.625) (layers F.Cu F.Paste F.Mask))
+  (model ${KISYS3DMOD}/Package_DFN_QFN.3dshapes/Diodes_UDFN-10_2.5x1.0mm_P0.5mm.wrl
```
Comment (chschlue) [inline Package_DFN_QFN.pretty/Diodes_UDFN-10_1.0x2.5mm_P0.5mm.kicad_mod]: Fix 3D model name (WxH)

**Comment by eclecticc** [comment]
Thanks, corrected.

**Comment by chschlue** [comment]
Thanks.

**Pull request merged**